Context propagation documentation (#9554)
micronaut-projects/micronaut-core: src/main/docs/guide/contextPropagation/httpFilterContextPropagation.adoc

> Modifying the propagated context is a common scenario, usually you want to extend the context to include the request related values.

Reviewer: ```suggestion
Modifying the propagated context is a common scenario. Usually, you want to extend the context to include the request-related values.
```

> To use a non-reactive HTTP filter API you need to add a method parameter api:core.propagation.MutablePropagatedContext[] and modify the propagated context elements by adding or removing the existing once:

Reviewer: ```suggestion
To use a non-reactive HTTP filter API, you need to add a method parameter api:core.propagation.MutablePropagatedContext[] and modify the propagated context elements by adding or removing the existing ones:
```

> snippet::io.micronaut.docs.propagation.MdcFilter[tags="class", indent=0, title="Example of adding a new MDC propagated context element"]

Reviewer: ```suggestion
.Example of adding a new MDC propagated context element
[source,java]
----
include::{testsuitejava}/propagation/MdcFilter.java[tags="class", indent=0]
----
```

The next filter in the chain will have the new propagated context available. Any of the thread-local context elements will be set for the next filter or the controller method invocation.

> To use the legacy reactive HTTP filters simply modify and propagate the context bound to the next chain invocation:

Reviewer: ```suggestion
To use the legacy reactive HTTP filters, simply modify and propagate the context bound to the following chain invocation:
```

> snippet::io.micronaut.docs.propagation.MdcLegacyFilter[tags="class", indent=0, title="Example of adding a new MDC propagated context element for the reactive filter"]

Reviewer: ```suggestion
.Example of adding a new MDC propagated context element for the reactive filter:
[source,java]
----
include::{testsuitejava}/propagation/MdcLegacyFilter.java[tags="class", indent=0]
----
```
